File: Basics/Strings.java
/*NOTICE: return at later date */

public class Strings{
    public static void main(String[] args){
        //define first_name
        char[] first_name = {'J', 'o' , 'h' ,'n'};
        //define last_name
        char[] last_name = {'D', 'o', 'e'};

        String name;

        last_name[0] = 'B';
        //create name reference separate from first_name & last_name
        name = new String(first_name)+
               " "+
               new String(last_name);
        if(name.compareTo("John Boe")==0){
            System.out.println("Done!");
        }

        name = "";
        name += new String(first_name);
        name += new String(last_name);
        System.out.println(name);
    }
}

File: Basics/Strings.c
#include <stdio.h>
#include <string.h>

int main(){
    /*define first_name*/
    char* first_name = "John";
    /*define last_name*/
    char last_name[] = "Doe";

    char name[100];

    last_name[0] = 'B';
    sprintf(name, "%s %s", first_name, last_name);
    if(strncmp(name, "John Boe", 100) == 0){
        printf("Done!\n");
    }
    name[0] = '\0';
    strncat(name, first_name, 4);
    strncat(name, last_name, 20);
    printf("%s\n", name);
    return 0;
}